accord-core/src/main/java/accord/local/SafeCommand.java

```diff
@@ -71,6 +70,12 @@ public Command addListener(Command.DurableAndIdempotentListener listener)
         return update(Command.addListener(current(), listener));
     }
 
+    public void addAndInvokeListener(SafeCommandStore safeStore, TransientListener listener)
```

Comment:
could you move this into the InMemorySafeCommand so it's not part of the public api?

Reply:
I think either we should remove this and make callers manually invoke the listener or starting using it in `Barrier` in which case it should be part of the API. I don't really have a preference, but for now I am going to switch `Barrier` to use it.